server/helpers/findLendingHistory.js

```diff
@@ -0,0 +1,29 @@
+import jwt from 'jsonwebtoken';
+import models from '../database/models';
+import responseMessage from './responseMessage';
+
+const { LendingHistory } = models;
+
+const findLendingHistory = async (id, bookId, response) => {
+  try {
+    let bookDuration = '';
+    const lentBook = await LendingHistory.findOne({ where: { userId: id, bookId, type: 'borrowed' }, order: [['createdAt', 'DESC']], limit: 1 });
+    if (lentBook) {
+      const { durationToken, durationStatus } = lentBook.dataValues;
+      if (durationStatus !== 'expired') {
+        jwt.verify(durationToken, process.env.JWT_KEY, async (error) => {
+          if (error) {
+            bookDuration = 'book token expired';
+            await LendingHistory.update({ durationStatus: 'expired' }, { where: { userId: id, bookId } });
```

Comment:
Why are you updating the lending history. i thought your route is just to borrow a book. I think there should a route for the user to read a borrowed book. If the user tries to read a borrowed book, the route would check if the duration token and if is expired, it should update the LendingHistory table. What do you think?

Reply:
Thinking about it, you are also right, but if I borrow and I don't read, I feel both routes can do that

Reply:
yeah you're right

Reply:
Since Its a digital library, I don't think you need to know if the book is already borrowed because it is unlimited. Immediately the user successfully borrows the book, the session starts reading. Anytime the user logs in and requests to read a book, another route is called using fetch api which checks if the durationtoken is expired and updates the lending history table if it is.
That should probably be on another task though since this is focused on borrowing books

Reply:
I was earlier told to add this implementation by the TTL to update and check for expired books

Reply:
Okay. I will bring it up during the standup

Reply:
Updated accordingly, thank you for the observation